go-controller/pkg/ovn/endpoints.go

```diff
@@ -20,35 +25,38 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 			ep.Name, ep.Namespace)
 		return nil
 	}
-	tcpPortMap := make(map[int32]([]string))
-	udpPortMap := make(map[int32]([]string))
+	tcpPortMap := make(map[string]lbEndpoints)
+	udpPortMap := make(map[string]lbEndpoints)
 	for _, s := range ep.Subsets {
 		for _, ip := range s.Addresses {
 			for _, port := range s.Ports {
 				var ips []string
-				var portMap map[int32]([]string)
-				var ok bool
+				var portMap map[string]lbEndpoints
 				if port.Protocol == kapi.ProtocolUDP {
 					portMap = udpPortMap
 				} else if port.Protocol == kapi.ProtocolTCP {
 					portMap = tcpPortMap
 				}
-				if ips, ok = portMap[port.Port]; !ok {
+				if lbEps, ok := portMap[port.Name]; ok {
+					ips = lbEps.IPs
+				} else {
 					ips = make([]string, 0)
 				}
 				ips = append(ips, ip.IP)
-				portMap[port.Port] = ips
+				portMap[port.Name] = lbEndpoints{IPs: ips, Port: port.Port}
```

Comment:
Do we always have a port.Name?

Reply:
After some experiments, it looks like port.Name can be "" if there is just one port in service. But looks like golang is okay with a key being "".